azure/services/managedclusters/managedclusters.go

```diff
@@ -314,6 +315,10 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			existingMC.NetworkProfile.LoadBalancerProfile.EffectiveOutboundIPs = nil
 		}
 
+		// Avoid changing agent pool profiles through AMCP and just use the existing agent pool profiles
+		// AgentPool changes are managed through AMMP
+		managedCluster.AgentPoolProfiles = existingMC.AgentPoolProfiles
```

Comment:
Does this thread summarize the discussion around the above addition?

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1896#pullrequestreview-818380859

@alexeldeib are we good with this as-is?

(Everything else in the PR lgtm)

Reply:
@zmalik are you able to answer Jack's question above?

Reply:
yes, we agreed to send the AgentPoolProfiles as they come from Azure API and not use AMCP reconcile to update agentpoolprofiles but only use AMMP reconcile to update the agentpools. 

In the current solution, this can give us a corner race condition where while AMMP is adding an agentPool and AMCP doesn't get it from the Azure API and tries to delete it. This returns an error as per @alexeldeib but we need AKS team to tell us which error it would be. And return a transient error in that case. 
Right now it's returning a normal error and would reconcile again where Azure API will return the newly added agentpool.

I will open a separate PR to handle this corner case as soon as we know what error is returned.

Reply:
Thanks! Can you please open an issue and assign you & Ace for the corner case so we remember to circle back after the holiday?

Reply:
yes! created https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1937